src/app/Fake.FluentMigrator/FluentMigrator.fs

```diff
@@ -0,0 +1,311 @@
+﻿/// Contains functions to run FluentMigrator
+module Fake.FluentMigrator
+
+open System
+open System.IO
+open Fake.Core
+open FluentMigrator.Runner
+open FluentMigrator.Runner.Initialization
+open FluentMigrator.Runner.Announcers
+
+let private horizontalRule = "".PadRight(79, '-')
+
+type internal FakeAnnouncer() =
+    inherit Announcer()
+
+    override this.Say message =
+        base.Say(sprintf "[+] %s" message);
+
+    override this.Heading message =
+        Trace.trace horizontalRule
+        Trace.trace message
+        Trace.trace horizontalRule
+
+    override this.Emphasize message = 
+        Trace.traceImportant (sprintf "[+] %s" message)
+
+    override this.Error message =
+        Trace.traceError (sprintf "!!! %s" message)
+
+    override this.Write (message, escaped) =
+        Trace.log message
+
+/// MS SQL Server driver version
+type SqlServerVersion = 
+    | Default
+    | V2000
+    | V2005
+    | V2008
+    | V2012
+    | V2014
+    | CE
+
+/// Oracle database driver version
+type OracleVersion =
+    | Default
+    | Managed
+    | DotConnect
+
+/// Fluent Migrator SQL syntax provider
+type DatabaseProvider =
+    | SqlServer of version: SqlServerVersion
+    | Oracle of version: OracleVersion
+    | DB2
+    | Firebird
+    | HANA
+    | Jet
+    | MySql
+    | PostgreSQL
+    | SQLite
+
+/// Database connection configuration
+type DatabaseConnection =
+    ///Explicit connection string
+    | ConnectionString of connectionString: string * provider: DatabaseProvider
+    ///Connection string specified in application config file
+    | ConnectionStringFromConfig of name: string * configPath: string * provider: DatabaseProvider
+
+/// Fluent Migrator execution mode
+type MigrationRunningMode =
+    /// Execute migrations on the target database
+    | Execute of connection: DatabaseConnection
+    /// Execute migrations on the target database and script SQL to the output file
+    | ExecuteAndScript of connection: DatabaseConnection * outputFileName: string
+    /// Execute migrations in preview-only mode
+    | Preview of connection: DatabaseConnection
+    /// Create migration script
+    | Script of startVersion: int64 * outputFileName: string * provider: DatabaseProvider
+
+/// Database operation to execute
+type DatabaseTask =
+    | MigrateUp of mode: MigrationRunningMode * version: Option<int64>
+    | MigrateDown of mode: MigrationRunningMode * version: int64
+    | Rollback of mode: MigrationRunningMode * steps: int
+    | ListAppliedMigrations of connection: DatabaseConnection
+
+/// Fluent Migrator execution mode
+type MigrationOptions = {
+    Namespace: Option<string * bool>;
+    Profile: string;
+    Tags: seq<string>;
+    Timeout: int;
+    Context: System.Object;
+    TransactionPerSession: bool;
+    ProviderSwitches: string;
+    WorkingDirectory: string;
+    Verbose: bool
+}
+
+/// Default migration options
+let DefaultMigrationOptions = {
```

Comment:
Can this be private/internal?

Reply:
In the previous version this was actually used to simplify using the helper.

I can either leave it public and the helper can operate as before, or simply define new  functions that don't take in options and use these, is either one more in line with the new FAKE 5 standards?

Reply:
Yes. Using functions to modify the arguments is used a lot more throughout the api

Reply:
> In the previous version this was actually used to simplify using the helper.

Can you clarify how that simplifies the helper? Yes it might make the code just a bit shorter, but at the cost of knowing that this API works differently and at the cost of having to know how the default object is called. If the code as much shorter and therefore more compose-able I'm probably open to making an exception here.

Reply:
I took a browse through the `Fake.Sql.DacPac` package. I will update `FluentMigrator` to use an argument modifier function. 

I'm not sure why the default args in that package are public, but I will make this private and use an modifier function, that is a much cleaner implementation.